src/main/java/attendance/model/MonthlyAttendance.java

```diff
@@ -0,0 +1,70 @@
+package attendance.model;
+
+import attendance.util.DateUtils;
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class MonthlyAttendance {
+
+    private final Month month;
+    private final Crew crew;
+    private final List<Attendance> attendances;
+
+    public MonthlyAttendance(Month month, Crew crew, List<Attendance> attendances) {
+        this.month = month;
+        this.crew = crew;
+        this.attendances = attendances;
+    }
+
+    public AttendanceResult calculateAttendanceResultUntilDate(LocalDate endDate) {
+        validateMonth(endDate);
+        List<Attendance> result = new ArrayList<>();
+        for (int date = 1; date <= endDate.getDayOfMonth(); date++) {
+            LocalDate currentDate = LocalDate.of(endDate.getYear(), endDate.getMonth(), date);
+            if (isCloseDay(currentDate)) {
+                continue;
+            }
+            Attendance addAttendance = attendances.stream()
+                    .filter(attendance -> attendance.isAlreadyAttendance(crew, currentDate))
+                    .findFirst().orElse(Attendance.absent(crew, currentDate));
```

Comment:
Stream의 체이닝 메서드도 한 줄에 하나의 메서드만 호출한다면 조금 더 가독성이 좋아질 것 같아요. 😃

```suggestion
                    .findFirst()
                    .orElse(Attendance.absent(crew, currentDate));
```